File: src/main/java/com/eltonkola/crud/liquidsoap/TelnetClient.java
package com.eltonkola.radioz.liquidsoap;

import java.io.*;
import java.net.Socket;

public class TelnetClient {

    public static String ICECAST_SKIP = "icecast.skip";
    public static String ICECAST_START = "icecast.start";
    public static String ICECAST_STATUS = "icecast.status";
    public static String ICECAST_STOP = "icecast.stop";
    public static String ICECAST_AUTOSTART = "icecast.autostart";
    public static String ICECAST_METADATA = "icecast.metadata";
    public static String ICECAST_REMAINING = "icecast.remaining";

    public static String HARBOR5002_BUFFER_LENGTH = "harbor_5002.buffer_length";
    public static String HARBOR5002_KICK = "harbor_5002.kick";
    public static String HARBOR5002_STATUS = "harbor_5002.status";
    public static String HARBOR5002_TOP = "harbor_5002.stop";

    public static String JINGLES_STOP = "jingles.stop";
    public static String JINGLES_NEXT = "jingles.next";
    public static String JINGLES_RELOAD = "jingles.reload";
    public static String JINGLES_URI = "jingles.uri ";  // + <URI>

    public static String REQUEST_ALIVE = "request.alive";
    public static String REQUEST_ALL = "request.all";
    public static String REQUEST_METADATA = "request.metadata "; // <rid>
    public static String REQUEST_ONAIR = "request.on_air";
    public static String REQUEST_RESOLVING = "request.resolving";
    public static String REQUEST_TRACE = "request.trace "; //+<rid>

    public static String SONGS_NEXT = "songs.next";
    public static String SONGS_RELOAD = "songs.reload";
    public static String SONGS_URI = "songs.uri "; // + [<URI>]

    public static String VAR_GET = "var.get ";//+<variable>
    public static String VAR_LIST = "var.list";
    public static String VAR_SET = "var.set ";// +<variable> = <value>

    public static String UPTIME = "uptime";
    public static String LIST = "list";
    public static String VERSION = "version";

    public static void main(String[] args) {
        TelnetClient ex = new TelnetClient();
        ex.send(ICECAST_SKIP);
    }

    public void send(final String cmd) {

        Socket pingSocket = null;
        PrintWriter out = null;
        BufferedReader in = null;

        try {

            pingSocket = new Socket("127.0.0.1", 1360);
            out = new PrintWriter(pingSocket.getOutputStream(), true);
            in = new BufferedReader(new InputStreamReader(pingSocket.getInputStream()));
            out.println(cmd);
            System.out.println(in.readLine());


        } catch (Exception e) {
            e.printStackTrace();
        }finally {
            try {
                if (out != null) {
                    out.close();
                }
                if (in != null) {
                    in.close();
                }
                if (pingSocket != null) {
                    pingSocket.close();
                }
            }catch (Exception ex){
                ex.printStackTrace();
            }
        }

    }

    public String getResponse(String cmd){

        Socket pingSocket = null;
        PrintWriter out = null;
        BufferedReader in = null;

        try {

            pingSocket = new Socket("127.0.0.1", 1360);
            out = new PrintWriter(pingSocket.getOutputStream(), true);
            in = new BufferedReader(new InputStreamReader(pingSocket.getInputStream()));
            out.println(cmd);
            String resposne = in.readLine();
            System.out.println(resposne);
            return resposne;

        } catch (Exception e) {
            e.printStackTrace();
        }finally {
            try {
                if (out != null) {
                    out.close();
                }
                if (in != null) {
                    in.close();
                }
                if (pingSocket != null) {
                    pingSocket.close();
                }
            }catch (Exception ex){
                ex.printStackTrace();
            }
        }
        return null;
    }



}

File: src/main/kotlin/com/eltonkola/cingerdhing/liquidsoap/TelnetClient.kt
package com.eltonkola.cingerdhing.liquidsoap

import java.io.*
import java.net.Socket

class TelnetClient {

    val ICECAST_SKIP = "icecast.skip"
    val ICECAST_START = "icecast.start"
    val ICECAST_STATUS = "icecast.status"
    val ICECAST_STOP = "icecast.stop"
    val ICECAST_AUTOSTART = "icecast.autostart"
    val ICECAST_METADATA = "icecast.metadata"
    val ICECAST_REMAINING = "icecast.remaining"

    val HARBOR5002_BUFFER_LENGTH = "harbor_5002.buffer_length"
    val HARBOR5002_KICK = "harbor_5002.kick"
    val HARBOR5002_STATUS = "harbor_5002.status"
    val HARBOR5002_TOP = "harbor_5002.stop"

    val JINGLES_STOP = "jingles.stop"
    val JINGLES_NEXT = "jingles.next"
    val JINGLES_RELOAD = "jingles.reload"
    val JINGLES_URI = "jingles.uri "  // + <URI>

    val REQUEST_ALIVE = "request.alive"
    val REQUEST_ALL = "request.all"
    val REQUEST_METADATA = "request.metadata " // <rid>
    val REQUEST_ONAIR = "request.on_air"
    val REQUEST_RESOLVING = "request.resolving"
    val REQUEST_TRACE = "request.trace " //+<rid>

    val SONGS_NEXT = "songs.next"
    val SONGS_RELOAD = "songs.reload"
    val SONGS_URI = "songs.uri " // + [<URI>]

    val VAR_GET = "var.get "//+<variable>
    val VAR_LIST = "var.list"
    val VAR_SET = "var.set "// +<variable> = <value>

    val UPTIME = "uptime"
    val LIST = "list"
    val VERSION = "version"


    fun send(cmd: String) {

        var pingSocket: Socket? = null
        var out: PrintWriter? = null
        var bufferedReader : BufferedReader? = null

        try {

            pingSocket = Socket("127.0.0.1", 1360)
            out = PrintWriter(pingSocket.getOutputStream(), true)
            bufferedReader = BufferedReader(InputStreamReader(pingSocket.getInputStream()))
            out.println(cmd)
            println(bufferedReader.readLine())


        } catch (e: Exception) {
            e.printStackTrace()
        } finally {
            try {
                if (out != null) {
                    out.close()
                }
                if (bufferedReader != null) {
                    bufferedReader.close()
                }
                if (pingSocket != null) {
                    pingSocket.close()
                }
            } catch (ex: Exception) {
                ex.printStackTrace()
            }

        }

    }

    fun getResponse(cmd: String): String? {

        var pingSocket: Socket? = null
        var out: PrintWriter? = null
        var `in`: BufferedReader? = null

        try {

            pingSocket = Socket("127.0.0.1", 1360)
            out = PrintWriter(pingSocket.getOutputStream(), true)
            `in` = BufferedReader(InputStreamReader(pingSocket.getInputStream()))
            out.println(cmd)
            val resposne = `in`.readLine()
            println(resposne)
            return resposne

        } catch (e: Exception) {
            e.printStackTrace()
        } finally {
            try {
                if (out != null) {
                    out.close()
                }
                if (`in` != null) {
                    `in`.close()
                }
                if (pingSocket != null) {
                    pingSocket.close()
                }
            } catch (ex: Exception) {
                ex.printStackTrace()
            }

        }
        return null
    }






}

//fun main(args: Array<String>) {
//    val ex = TelnetClient()
//    ex.send(ex.ICECAST_SKIP)
//}
